miio/yeelight.py

```diff
@@ -312,63 +331,163 @@ def on(self, transition=0, mode=0):
         4: color flow
         5: moonlight
         """
-        if transition > 0 or mode > 0:
-            return self.send("set_power", ["on", "smooth", transition, mode])
-        return self.send("set_power", ["on"])
+        if (
+            lamp == YeelightSubLightType.Background
+            and not self.model_specs.has_background_light
+        ):
+            raise YeelightException("Background lamp not supported")
+
+        miio_command = SUBLIGHT_PROP_PREFIX[lamp] + "set_power"
+        miio_param = ["on"]
+        if transition > 0:
+            miio_param += ["smooth", transition]
+        else:
+            miio_param += ["sudden", 0]
```

Comment:
Transition seems to be used in several places, so it'd be better to add an internal method that performs its handling.